tools/parse/src/main/scala/org/allenai/nlpstack/parse/poly/core/SentenceTransform.scala

```diff
@@ -161,9 +164,106 @@ case class VerbnetTagger(verbnet: Verbnet) extends SentenceTransform {
-      val tokVerbnetSecondaryFrames = verbnet.getVerbnetFrameSecondaryNames(tokLemmaLC)
       tok.updateProperties(Map('verbnetPrimaryFrames -> tokVerbnetPrimaryFrames))
     })
   }
 }
 
+/** Frequency Distribution of dependency labels for tokens based on Google Ngram's Nodes (unigrams).
+  */
+case class GoogleUnigramDepLabelTagger(googleNgram: GoogleNGram) extends SentenceTransform {
+
+  @transient private val stanfordTagger = new StanfordPostagger()
+
+  override def transform(sentence: Sentence): Sentence = {
+    val taggedTokens = SentenceTransform.getPostaggedTokens(sentence, defaultPostagger)
+    Sentence(NexusToken +: (taggedTokens.zip(sentence.tokens.tail) map {
+      case (tagged, untagged) =>
+        val depLabelFreqMap = GoogleUnigram.getDepLabelNormalizedDistribution(
+          tagged, googleNgram.ngramMap, googleNgram.frequencyCutoff
+        )
+        // Create feature for each dependency label based on the normalized frequency
+        // bucket it lies in.
+        val frequencyFeatureMap: Map[Symbol, Set[Symbol]] = (for {
+          depLabel <- depLabelFreqMap.keySet
+        } yield {
+          val normalizedFrequency = depLabelFreqMap(depLabel)
+          val symbolSetWithCurrentDepLabel = Set(Symbol(depLabel))
+          if (normalizedFrequency <= 0.0009) {
+            ('depLabelFreqSmall, symbolSetWithCurrentDepLabel)
+          } else if (normalizedFrequency <= 0.01) {
+            ('depLabelFreqBelow1, symbolSetWithCurrentDepLabel)
+          } else if (normalizedFrequency <= 0.1) {
+            ('depLabelFreq1to10, symbolSetWithCurrentDepLabel)
+          } else if (normalizedFrequency <= 0.2) {
+            ('depLabelFreq11to20, symbolSetWithCurrentDepLabel)
+          } else if (normalizedFrequency <= 0.3) {
+            ('depLabelFreq21to30, symbolSetWithCurrentDepLabel)
+          } else if (normalizedFrequency <= 0.4) {
+            ('depLabelFreq31to40, symbolSetWithCurrentDepLabel)
+          } else if (normalizedFrequency <= 0.5) {
+            ('depLabelFreq41to50, symbolSetWithCurrentDepLabel)
+          } else if (normalizedFrequency <= 0.6) {
+            ('depLabelFreq51to60, symbolSetWithCurrentDepLabel)
+          } else if (normalizedFrequency <= 0.7) {
+            ('depLabelFreq61to70, symbolSetWithCurrentDepLabel)
+          } else if (normalizedFrequency <= 0.8) {
+            ('depLabelFreq71to80, symbolSetWithCurrentDepLabel)
+          } else if (normalizedFrequency <= 0.9) {
+            ('depLabelFreq81to90, symbolSetWithCurrentDepLabel)
+          } else {
+            ('depLabelFreq91to100, symbolSetWithCurrentDepLabel)
+          }
+        }).groupBy(_._1).mapValues(_.flatMap(v => v._2))
+        untagged.updateProperties(frequencyFeatureMap)
+    }))
+  }
+}
+
+/** Frequency Distribution of POS tags for words based on Google Ngram's Nodes (unigrams).
+  */
+case class GoogleUnigramPostagTagger(googleNgram: GoogleNGram) extends SentenceTransform {
```

Comment:
There's a lot of code duplication between this and the previous class. Is there a way to refactor?